### DISCUSSION TITLE
Replay_memory.sample qué uso tiene

### DISCUSSION THREAD

**Discussion by valeria-adasme** [JavierBZ/Tarea3-IIC3697#21] [Q&A]
Tengo una duda muy básica. Había entendido que la función 'sample' de la memoria era para sacar un 'batch_size' número de muestras. Pero al ver el uso que se le da en el código ejemplo de training, creo que me pierdo un poco ¿Lo que está haciendo en la línea

`obs_t, act_t, rew_t, next_obs, done_mask = replay_memory.sample(batch_size)`

del training es entregar un vector obst_t (por ejemplo) de largo 1x32?

**Comment by JavierBZ** [comment]
La función sample "saca" de la memoria 32 experiencias que usarás para calcular loss y hacer un paso de optimización. 
obs_t no puede ser 1x32, son tus estados, deberían ser 32x4x84x84.
Algunos de los demas como act_t, rew_t  son de 32x1
